### PR TITLE
dashboard/feature/implement_taxi_search_with_filter

### PR DESCRIPTION
**Pull request by Asiasama710**
- i implement get taxi with search and filter logic

### CODE REVIEW

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/ManageTaxisUseCase.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ interface IManageTaxisUseCase {
     suspend fun createTaxiReport()
 
     suspend fun getTaxis(
-        username: String? = null,
+        username: String?,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/ManageTaxisUseCase.kt:18]: why username is nullable if it's getting all taxis when it's nullable and by username when it's not null this fucntion should be seprated into two functions

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/TaxiFiltration.kt [hunk 2]
```diff
@@ -5,5 +5,5 @@ import org.thechance.common.domain.util.TaxiStatus
 data class TaxiFiltration(
     val carColor: CarColor?,
     val seats: Int?,
-    val status: TaxiStatus?,
+    val status: TaxiStatus?=null,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/TaxiFiltration.kt:8]: why give it initial value ?

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiScreenModel.kt [hunk 3]
```diff
@@ -33,13 +33,12 @@ class TaxiScreenModel(
...
-                    state.value.taxiFilterUiState.toEntity(),
-                    state.value.currentPage,
-                    state.value.specifiedTaxis
-                )
+            { manageTaxis.getTaxis(
+                username=  mutableState.value.searchQuery.trim(),
+                taxiFiltration= mutableState.value.taxiFilterUiState.toEntity(),
+                page=  mutableState.value.currentPage,
+                limit=  mutableState.value.specifiedTaxis
+            )
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiScreenModel.kt:41]: try better way 
For example mutableState.value.apply{
}

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiUiState.kt [hunk 4]
```diff
@@ -61,7 +61,7 @@ data class TaxiDetailsUiState(
 
 data class TaxiFilterUiState(
     val carColor: CarColor? = null,
-    val seats: Int = -1,
+    val seats: Int? = null,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiUiState.kt:64]: don't give null values in ui state give initial values

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiUiState.kt [hunk 5]
```diff
@@ -61,7 +61,7 @@ data class TaxiDetailsUiState(
...
     val status: TaxiStatus? = null
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiUiState.kt:65]: don't give null values in ui state give initial values

**Review by aya-seif — APPROVED** [review]

**Review by hassanwasfy — APPROVED** [review]

**Review by AndrewAboalhana — APPROVED** [review]

**Review by devsadeq — APPROVED** [review]

**Review by ahmednasserzaza — APPROVED** [review]

**Pull request merged**